clickstream-event-visualiser/src/main/java/com/gojek/clickstream/clickstream_event_visualiser/EVEventRepository.kt

```diff
@@ -0,0 +1,41 @@
+package com.gojek.clickstream.clickstream_event_visualiser
+
+import clickstream.interceptor.InterceptedEvent
+
+/**
+ * Singleton class that handles intercepted event from [EventVisualiserInterceptor].
+ * Set [onNewEvent] to receive the updated events.
+ *
+ * */
+public object EVEventRepository {
+
+    /**
+     * Callback to receive updated events.
+     *
+     * */
+    public var onNewEvent: (List<EVEvent>) -> Unit = {}
+
+    private var listenToEvent = true
+
+    /**
+     * Enables event updates.
+     *
+     * */
+    public fun startListeningToEvent() {
+        listenToEvent = true
+    }
+
+    /**
+     * Disables event updates.
+     *
+     * */
+    public fun stopListeningToEvent() {
```

Comment:
just call the function `stop`? since the class name already representative what it does